enhancements/subscription-content/subscription-content-access.md

```diff
@@ -0,0 +1,242 @@
+
+#### Access Appropriate RHEL Content for Container
+
+As a developer using OpenShift I want to be able to access appropriate RHEL content for my container
+So that I download RHEL 8 content in UBI 8 containers And download RHEL 7 content in UBI 7
+containers
+
+### Implementation Details/Notes/Constraints [optional]
+
+To consume RHEL content in OpenShift, the running container must have access to the following:
+
+One or more entitlement keys, which are PEM-encoded certificates. A yum repository definition, which
+tells yum/dnf where to access content from.
+
+Yum repository definitions are specific to the major version of RHEL that the running container is
+based on - a RHEL 8-based container is not compatible with RHEL 7 content.
+
+This enhancement proposal will detail at a high level the functions of:
+
+A cluster-wide subscription manager to generate subscription content access data A projected
+resource CSI driver to share resources across OpenShift namespaces A subscription content webhook to
+generate the necessary volume mounts to add subscription content access data to containers
+Extensions to propagate subscription content access data into OpenShift builds.
+
+#### Cluster Subscription Manager
+
+The cluster subscription manager will take as input the `cloud.redhat.com` pull secret, which
+already provides authentication for the cluster and provides the means to register clusters with Red
+Hat’s cloud manager. The pull secret is then used to obtain the following: The `redhat.repo` yum
+repo configurations for all supported major versions of RHEL. Today these will be RHEL 7 and 8 -
+later major versions of RHEL must be anticipated. The Simple Content Access certificate for the
+customer, which will serve as the entitlement key.
+
+As RHSM does today on RHEL hosts, the cluster subscription manager will be responsible for
+generating `redhat.repo` definitions for the currently supported RHEL major versions and saving it
+on the cluster. Each `redhat.repo` file will be saved in a ConfigMap within the `openshift-config`
+namespace, named `redhat-repo-rhel${n}`, where `${n}` is the major version of RHEL associated with
+the `redhat.repo` file.
+
+The cluster subscription manager will also need to obtain the entitlement keys associated with the
+attached OpenShift subscription. It is assumed that OpenShift entitlements grant access to RHEL
+content, but may not grant access to layered content such as EAP or Application Runtimes. This
+content must be saved in a Secret within the `openshift-config` namespace, named
+`etc-pki-entitlement`.
+
+The full details of this component will be specified in a subsequent enhancement proposal.
+
+#### Projected-resource CSI Driver
+
+Once the entitlement key(s) and yum repository definitions are stored on the cluster, the next step
+is to share these definitions across namespaces. This task will fall to the projected-resource
```

Comment:
make `projected-resource ...` a link to https://github.com/openshift/enhancements/blob/master/enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

Reply:
will do on the next push.